### DISCUSSION TITLE
Model with three unique index referencing optional fields

### DISCUSSION THREAD

**Discussion by JohannesKlauss** [prisma/prisma#11201] [Q&A] [answered]
I got a question regarding unique indexes with optional values.
I have a schema like this:

```prisma
model Thread {
  id               Int            @id @default(autoincrement())
  createdAt        DateTime       @default(now())
  updatedAt        DateTime       @updatedAt
  body             String
  user             User           @relation(fields: [userId], references: [id])
  userId           Int
  likes            Like[]         @relation("ThreadsOnLikes")
}

model Like {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  user      User     @relation(fields: [userId], references: [id])
  userId    Int
  thread    Thread?   @relation("ThreadsOnLikes", fields: [threadId], references: [id])
  threadId  Int?
  comment   Comment? @relation("CommentsOnLikes", fields: [commentId], references: [id])
  commentId Int?

  @@unique([userId, threadId, commentId])
}
```

And inside a resolver I want to for example delete a like from a user for a specific threadId like so:

```js
await db.thread.update({
    where: { id: input.id },
    data: {
      likes: {
        delete: {
          userId_threadId_commentId: { userId: session.userId, threadId: input.id },
        },
      }
    }
  })
```

But when I try to execute that mutation, prisma throws the following error:
`Argument commentId for data.likes.delete.userId_threadId_commentId.commentId is missing.`
When I add it to the delete argument with , commentId: null  it states this error:
`Argument commentId: Got invalid value null on prisma.updateOneThread. Provided null, expected Int.`
Although inside the database the comment_id field is actually null . Is this a bug or how is this fixable?

**Comment by austincrim** [comment, marked as answer]
Hey @JohannesKlauss 👋 ,

Per [our docs on `@@unique`](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#unique-1):

> All fields that make up the unique constraint must be mandatory fields. The reason for this behavior is that all connectors consider null values to be distinct, which means that two rows that look identical are considered unique

Since `threadId` and `commentId` are marked as optional in your schema, what you are trying to accomplish is not support by Prisma.

**Reply by JohannesKlauss** [reply]
Hmh, but shouldn't prisma throw an error then while validating the schema?

**Reply by austincrim** [reply]
Absolutely! Would you mind filling out a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with your specific environment info and use case? Our engineers will pick it up from there 👍.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
